crates/shared/src/transport/http.rs

```diff
@@ -147,6 +147,12 @@ impl BatchTransport for HttpTransport {
 
         async move {
             let _guard = metrics.on_request_start("batch");
+            calls.iter().for_each(|call| {
+                metrics
+                    .inner_batch_requests_complete
+                    .with_label_values(&[method_name(call)])
+                    .inc()
+            });
```

Comment:
nit: Technically the requests only completed after `execute_rpc()` in line 157. That shouldn't matter in the grand scheme of things, though.

Reply:
It is unclear whether we want to measure before or after. If we measure before then it is possible that the request is dropped before the node handles it. If we measure after then it is possible that the node handled but we dropped it before receiving. If the goal is to measure number of calls for pricing then I'm slightly leaning to increment before. Could rename the metric to "initiated" instead of "complete". Anyway, it's a small detail.

Reply:
Yes indeed, the goal is to measure calls for pricing.

> Could rename the metric to "initiated" instead of "complete"

Sound's good. Will update.